platform-sdk/docs/proposals/DP-0x/0X - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,498 @@
+- **Groth 21**:publicly verifiable secret sharing and resharing schemes that enable secure and efficient distribution and management of secret shares, with many possible uses cases supporting applications in distributed key generation and threshold signatures. Uses Shamir's secret sharing and ElGamal.
+- **Shamir’s Secret Sharing**: In Shamir’s SS, a secret s is divided into n shares by a dealer, and shares are sent to shareholders secretly. The secret "s" is shared among n shareholders in such a way that
+(a) t or more than t shares can be used to recover the secret, and
+(b) fewer than t shares are not sufficient to obtain the secret
+Shamir’s (t, n) SS is based on a linear polynomial and is unconditionally secure.
+- **ElGamal**: On a message and a random number chosen by a signer, ElGamal signature scheme produces a signature consisting of two numbers (r, s),
+where r is computed from a random number regardless of a message and then s is computed from a message,
+a signer's secret key, a chosen random number, and r.
+- **EC (Elliptic Curve)**: "Elliptic" is not elliptic in the sense of an "oval circle". In the field "Fp", an EC is like a non-connected cloud of points. All points satisfy an equation and all operations are performed modulo p. Some elliptic curves are pairing-friendly.
+- **Bilinear Pairings**: Are mathematical functions used in cryptography to map two elements of different groups (in EC, the group is an elliptic curve) to a single value in another group in a way that preserves certain algebraic properties.
+- **Fields**: Mathematical structures where addition, subtraction, multiplication, and division are defined and behave as expected (excluding division by zero).
+- **Groups**: Sets equipped with an operation (like addition or multiplication) that satisfies certain conditions (closure, associativity, identity element, and inverses).
+- **Share**: Represents a piece of the public/private necessary elements to create signatures. In TSS, a threshold number of shares is needed to produce an aggregate signature that can be then verified by the ledger public key.
+- **Polynomial Commitment**: A polynomial commitment is a cryptographic commitment to a polynomial. It enables a committer to commit to a polynomial such that the commitment allows verifying evaluations of the polynomial at specific points without revealing the entire polynomial.
+
+### Goals
+- **Usability:** Design a user-friendly library with a public API that is easy to integrate with other projects like consensus-node and block-node.
+- **EVM support:** Generated signature and public keys should be compatible with evm precompiled functions so that signature validation can be done on smart contracts without incurring in excessive gas cost. 
```

Comment:
```suggestion
- **EVM support:** Generated signature and public keys should be compatible with evm precompiled functions so that signature validation can be done on smart contracts without incurring an excessive gas cost. 
```